### ISSUE TITLE
USB keyboard becomes unresponsive during recover

### ISSUE THREAD

**Issue by ChrisGet** [rear/rear#1870]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* ReaR version ("/usr/sbin/rear -V"): 
2.4 / Git

* OS version ("cat /etc/rear/os.conf" or "lsb_release -a" or "cat /etc/os-release"):
OS_VENDOR=Debian
OS_VERSION=8

* ReaR configuration files ("cat /etc/rear/site.conf" or "cat /etc/rear/local.conf"):
# Default is to create Relax-and-Recover rescue media as ISO image
# set OUTPUT to change that
# set BACKUP to activate an automated (backup and) restore of your data
# Possible configuration values can be found in /usr/share/rear/conf/default.conf
#
# This file (local.conf) is intended for manual configuration. For configuration
# through packages and other automated means we recommend creating a new
# file named site.conf next to this file and to leave the local.conf as it is. 
# Our packages will never ship with a site.conf.

OUTPUT=ISO
BACKUP=NETFS
BACKUP_URL="nfs://192.168.0.5/path/to/my/share/"

* Real hardware (PC or PowerNV BareMetal or ARM) and/or virtual machine (KVM guest or PoverVM LPAR):
PC/Server - Dell PowerEdge R210 II

* System architecture (x86 compatible or POWER PPC64/PPC64LE or what excat ARM device):
x86

* Are you using BIOS or UEFI or another way to boot (Open Firmware or Petitboot)?
Unsure

* Brief description of the issue:
Booting from the recovery iso that was made during "mkrescue" works. I can select auto recover option and then output gets stuck on "The original network interface Eth0 [MAC Address] is not available"

The 2 other network interfaces are listed with the numbers to select them but pressing the number keys on the keyboard does not register. In fact no keys on the keyboard register. The backup was done on a headless server machine via ssh.

* Work-around, if any:
None found so far.

**Comment by didacog** [comment]
Hi @ChrisGet,

We also had problems sometimes with some USB keyboards and we solved them using:
MODULES=( 'all_modules' ) in rear config (/etc/rear/local.conf).

You'll need to re-run backup with this variable in place. I hope this solve your issue with the keyboard detection.

Regards,
Didac

**Comment by ChrisGet** [comment]
Thanks! I will try again and let you know how it goes

**Comment by ChrisGet** [comment]
This fixed the keyboard problem. Thank you very much.

**Comment by jsmeix** [comment]
It seems this one is basically the same as
https://github.com/rear/rear/issues/1860

In both cases a Debian system is used.
Perhaps on Debian something is different that results
that at least on Debian 8 and 9 a USB keyboard
does not "just work" in the ReaR recovery system?

I cannot help with Debian specific issues because I do not use it.

**Comment by jsmeix** [comment]
@didacog 
when you had already a few of such issues
perhaps you could imagine if there is something in common
under what circumstances those issues happens which could help us
to enhance ReaR so that a USB keyboard always "just works"
because nowadays that is basically mandatory.

**Comment by ChrisGet** [comment]
Yes sorry I saw the other bug once I closed this one. I did look through existing issues before raising this one but didn't think to check the closed bugs. Apologies for the duplicate issue. I would have found my answer on #1860

**Comment by jsmeix** [comment]
@ChrisGet 
what I am most interested is what the actual root cause is why sometimes
a USB keyboard does not "just work" in the ReaR recovery system.
If you have an idea or if you could dig into it you could help much.

**Comment by didacog** [comment]
@jsmeix 

I had this issue 2/3 times previously and I guessed that some module was not loading in rescue. I used  'all_modules' to solve it, and I didn't have time to go deeper onto the root problem. I think it was also with Debian, then we have now an starting point to see whats happening. Maybe @ChrisGet can try it again and check the modules loaded in the rescue image with and without 'all_modules' set and we can try to figure out which module is missing. 

This does not happen all times, I've recovered lots of debian systems without issue...
